### ISSUE TITLE
Custom attribute blob parser issue

### ISSUE THREAD

**Issue by CursedSheep** [Washi1337/AsmResolver#109]
ASMresolver fails to write this sample file
[lel2.zip](https://github.com/Washi1337/AsmResolver/files/5596292/lel2.zip)

code used:
```cs
        static void Main(string[] args)
        {
            try
            {
                filename = args[0];
            }
            catch
            {
                Console.Write("Enter file: ");
                filename = Console.ReadLine().Replace(@"""", "");
            }
            ModuleDefinition module = ModuleDefinition.FromFile(filename);
            var imageBuilder = new ManagedPEImageBuilder();
            var factory = new DotNetDirectoryFactory();
            factory.MetadataBuilderFlags = MetadataBuilderFlags.PreserveBlobIndices
                                         | MetadataBuilderFlags.PreserveTypeReferenceIndices;
            factory.MethodBodySerializer = new CilMethodBodySerializer
            {
                ComputeMaxStackOnBuildOverride = false
            };
            imageBuilder.DotNetDirectoryFactory = factory;
            string patho = Path.GetDirectoryName(filename) + "\\" + Path.GetFileNameWithoutExtension(filename) + "-test" + Path.GetExtension(filename);
            module.Write(patho, imageBuilder); 
            Console.WriteLine("saved to " + patho);
            Console.ReadKey();
        }
```
Exception info:
`
Exception thrown: 'System.AggregateException' in AsmResolver.DotNet.dll
An unhandled exception of type 'System.AggregateException' occurred in AsmResolver.DotNet.dll
Construction of the PE image failed with one or more errors.
Inner Exception: FormatException: Expected one of CloseBracket, Comma, Number.
`

**Comment by Washi1337** [comment]
I just pushed a potential fix on branch [issue/109](https://github.com/Washi1337/AsmResolver/tree/issue/109). Can you verify that this indeed fixes your problem?

**Comment by Washi1337** [comment]
Fixed in 4.2
